libknet/compress_lz4hc.c

```diff
@@ -22,8 +23,13 @@
 #define KNET_LZ4HC_MAX LZ4HC_MAX_CLEVEL
 #endif
 #ifndef KNET_LZ4HC_MAX
-#define KNET_LZ4HC_MAX 0
-#error Please check lz4hc.h for missing LZ4HC_CLEVEL_MAX or LZ4HC_MAX_CLEVEL variants
+/*
+ * older releases of lz4 do not define LZ4HC_CLEVEL range.
+ * According to lz4hc.h, any value between 0 and 16 is valid.
+ * We defalt to 9 based on the comments in the include file
+ * from older versions.
+ */
+#define KNET_LZ4HC_MAX 9
```

Comment:
This comment is a little confusing, do you set 9 as default or as maximum? Or is it that knet defaults to the maximum allowed by the compressor?
Also, 0 means "use default value" for LZ4, but "no compression" for zlib (-1 selects some default there), which knet does not allow. Do we need consistency here?